galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/common/PropertyUtilities.java

```diff
@@ -0,0 +1,243 @@
+         * Therefore setting updateProperty to false will force a create property path,
+         * whilst setting updateProperty to true will force an update property path
+         */
+        if (propExists == updateProperty){
+            getFramework().getConfigurationPropertyService(property.metadata.namespace).setProperty(property.metadata.name, property.data.value);
+        }else if (propExists){
+            ServletError error = new ServletError(GAL5018_PROPERTY_ALREADY_EXISTS_ERROR, property.metadata.name, property.metadata.namespace);  
+            throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+        }else{
+            ServletError error = new ServletError(GAL5017_PROPERTY_DOES_NOT_EXIST_ERROR, property.metadata.name);  
+            throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+        }
+    }
+
+    public void setGalasaProperty (GalasaProperty property, String action) throws FrameworkException{
+        boolean updateProperty = false;
+        if (isPropertyValid(property) && updateActions.contains(action)){
+            if ((checkGalasaPropertyExists(property) || action.equals("update"))){
+                updateProperty = true;
+            }
+        }
+        setProperty(property, updateProperty);
+    }
+
+    public boolean checkPropertyNamespaceMatchesURLNamespace(@NotNull GalasaProperty property , @NotNull String namespace){
+        return namespace.toLowerCase().trim().equals(property.metadata.namespace.toLowerCase().trim());
+
+    }
+     /**
+     * Returns an GalasaProperty from the request body that should be encoded in UTF-8 format
+     * @param request
+     * @return GalasaProperty 
+     * @throws IOException
+     * @throws InternalServletException
+     */
+    public GalasaProperty getPropertyFromRequestBody (HttpServletRequest request) throws IOException, InternalServletException{
+        String body = new String (request.getInputStream().readAllBytes(),StandardCharsets.UTF_8);
+        return getGalasaPropertyfromJsonString(body);
+    }
+
+    /**
+     * This function casts a json String into a GalasaProperty so that it can be used by the framework.
+     * The property Json Structure should match the GalasaProperty Structure, otherwise an exception will be thrown
+     * 
+     * @param jsonString
+     * @return GalasaProperty
+     * @throws InternalServletException
+     */
+    public GalasaProperty getGalasaPropertyfromJsonString (String jsonString) throws InternalServletException{
+        GalasaProperty property = null;
+        boolean valid = false;
+        try {
+            property = gson.fromJson(jsonString, GalasaProperty.class);
+            valid = property.isPropertyValid();
+        }catch (Exception e){}
+        
+        if(!valid){
+            ServletError error = new ServletError(GAL5023_UNABLE_TO_CAST_TO_GALASAPROPERTY, jsonString);  
```

Comment:
This error could be misleading if the property was serialized correctly in `gson.fromJson` but was invalid for another reason in `property.isPropertyValid()`